src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import Hark from 'hark';
+import PropTypes from 'prop-types';
+import Graphicequaliser from './canvasequalizer/canvasequalizer';
+import PlayButton from './playbutton/playbutton';
+import Uploadbutton from './upload/uploadbutton';
+import Infoabouttrack from './upload/infoaboutfile/infoaboutuploadfile';
+import Streambutton from './streambutton/streambutton';
+import { connect } from 'react-redux';
+import { createAudioData, playPauseSoundFromFile, createStreamData, startMuteStreamAudio, mergeCanvasWidth } from '../../actions/audioactions';
+
+
+class Equalizer extends React.Component {  
+   componentDidMount(){ 
+      this.detectStreamSoundFromMicrophon();
+    }
+   
+  detectStreamSoundFromMicrophon=()=>{    
+    const context = this.props.audioData.audiocontext;          
+    const audioLineIn = new Audio();   
+    if (navigator.mediaDevices) {      
+      navigator.mediaDevices.getUserMedia ({audio: true})      
+      .then((stream)=> {   
+        audioLineIn.srcObject = stream;
+        audioLineIn.muted=true;        
+          //hark
+          const options = {};
+          const speechEvents = Hark(stream, options);
+          const htmlinfo=document.getElementById("stream_detecting");      
+          speechEvents.on('speaking', ()=>{           
+            htmlinfo.innerHTML=`speaking`;
+          });
+       
+          speechEvents.on('stopped_speaking', ()=>{
+            htmlinfo.innerHTML=`no stream detekting`;
+          });
+          const sourceStream = context.createMediaStreamSource(stream);                           
+          this.props.createStreamData({audioLineIn: audioLineIn, sourceStream});          
+      })
+      .catch(function(e) {
+        throw new Error(e);
+      });
+  } else {
+    throw new Error('browser doesnt support audio API');
+  }  
+  }  
+
+  widthMerge=(e)=>{   
+    this.props.mergeCanvasWidth(e);
+  }     
+
+  playSoundFromFile=(e)=>{
+    const soundfromfile=this.props.audioData.audioFromFile;    
+    if (this.props.audioData.playpausestate===false) {     
```

Comment:
this code can be shortened,
1. `playpausestate` can be destructured
2. `if (playpausestate) {`
3. `this.props.playPauseSoundFromFile();` is called in both cases, so no need to repeat it, you can call this function after `if` structure

Reply:
done